Feat (Ems-11): Implement user login FE-BE

MennaAllahYasser11 opened the pull request:
- Implement user login interface 
 - Used Appwrite for backend 
- Used zod for validation and react toastify for error handling and login successfully.

MennaAllahYasser11 committed:
feat: add and implement user login with appwrite

MennaAllahYasser11 committed:
feat: add route for login page

AlyaaRushdy committed:
feat: forgot and reset password without error handling

Anasmoner2022 committed:
feat: implement registration loader and update navigation logic in Register component

Anasmoner2022 reviewed (changes requested):
1. Title of page --> "Welcome back to your platform" **{for Ex.}**.
2. Add pragraph below the title.
3. Add Forget Password link.
4. Remove "now" from title of Button or make it "Sign in". "Personal Opinion"
5. Add autoFocus to first element in form.
6. replace useEffect code with loader module in react-router library.
7. Add zod Schema .

Anasmoner2022 committed:
fix: improve code formatting and consistency in Register component and registerLoader

Anasmoner2022 committed:
feat: add LinkedIn icon and implement registration validation schema

Anasmoner2022 committed:
feat: add zodResolver for form validation and update dependencies

Anasmoner2022 committed:
feat: add react-toastify for user feedback and update registration logic

Anasmoner2022 committed:
feat: add react-toastify for user feedback and update registration logic

Anasmoner2022 committed:
Merge branch 'auth' of https://github.com/web-hawks/evento into auth

MennaAllahYasser11 committed:
feat EMS-11: add route and loader for login

MennaAllahYasser11 committed:
feat EMS-11: add loader function and edit title and style

MennaAllahYasser11 committed:
feat EMS-11: add login schema for validation

MennaAllahYasser11 committed:
feat EMS-11: implement loader function for login

AlyaaRushdy committed:
Merge branch 'master' into EMS-12-forget-password-feature

Anasmoner2022 reviewed (approved):
Nice Work
But try to change your pull request name to be conventional .

Anasmoner2022 committed:
feat: update favicon, add new social media icons, remove old banner, and modify Tailwind CSS color palette

Anasmoner2022 committed:
Merge branch 'auth' into EMS-19-setup-project-assets

Anasmoner2022 committed:
feat: add GIF banner and light and dark logo

Anasmoner2022 committed:
style : change color of some texts

AlyaaRushdy committed:
feat: forgot and reset password error handling with toast

AlyaaRushdy committed:
refactor: replace manual validation with Zod schemas

mohamed17-a committed:
Merge remote-tracking branch 'origin/auth' into EMS-13-OAuth-setup

AlyaaRushdy committed:
chore: updated the logo and extracted the logo link a new component

MennaAllahYasser11 committed:
feat(EMS-11): add toastify and zodResolver

MennaAllahYasser11 committed:
feat(EMS-11): Edit validation for password

AlyaaRushdy committed:
refactor: refactor form fields into a map and some minor changes

MennaAllahYasser11 committed:
feat (EMS-11): Edit path of forget password

AlyaaRushdy reviewed (changes requested):
Great work as usual 👏.
please consider making these modifications:
- add more padding between the elements
- make the welcome message more user friendly
- add a bit of horizontal padding to the welcome message
- change "forget your password" to "forgot your password"
- make the OAuth section match the one in the registration page
- make the "already have an account" section match the one in the registration page.

AlyaaRushdy committed:
chore: update the bg color on darkmode and make some changes to the OAuth buttons style

AlyaaRushdy committed:
chore: update the bg color on darkmode, make some changes to the OAuth buttons style and some other minor changes

AlyaaRushdy committed:
Merge branch 'EMS-19-setup-project-assets' of https://github.com/web-hawks/evento into EMS-19-setup-project-assets

Anasmoner2022 committed:
Merge branch 'EMS-11-Login-feature' into auth

Anasmoner2022 committed:
feat, refactor, style, fix: Improved user experience and security enhancements

- Style: Updated color of "Password Strength" in dark mode.
- Feat: Added navigation to home page after successful registration.
- Feat: Prevented access to registration page when logged in using route loader.
- Refactor: Extracted OAuth section into a separate component.
- Refactor: Refactored form code to use map for dynamic field generation.
- Fix: Customized toast messages for better user interaction.

Anasmoner2022 reviewed (changes requested):
please merge your branch with project assets branch to get colors and banner with last updates and i think if you make your login like that will be a good view(ignore the content :) ):
![لقطة شاشة 2025-01-10 154325](https://github.com/user-attachments/assets/5820a44a-75a4-4b71-a499-3b0e72734454)

Anasmoner2022 committed:
Merge branch 'EMS-12-forget-password-feature' into auth

MennaAllahYasser11 committed:
Merge branch 'EMS-19-setup-project-assets' into EMS-11-Login-feature

MennaAllahYasser11 committed:
Merge branch 'auth' into EMS-11-Login-feature

mohamed17-a committed:
Merge remote-tracking branch 'origin/auth' into EMS-13-OAuth-setup

mohamed17-a committed:
feat : add Google OAuth functionality

mohamed17-a committed:
feat: add Facebook OAuth functionality

MennaAllahYasser11 committed:
feat (EMS-11): refactor the code to be align with register page

AlyaaRushdy committed:
chore: add a disc paragraph in forget password page, add loader the routes and change forgot/reset password route path and createRecovery link

MennaAllahYasser11 committed:
feat (EMS-11): add link for forgot password

MennaAllahYasser11 committed:
feat (EMS-11): Edit name conventional and path of forgot password

AlyaaRushdy reviewed (approved):
please remove the word "here" in "log in here"

MennaAllahYasser11 committed:
feat(EMS-11): Edit heading of login

Anasmoner2022 reviewed (approved):
Good work

MennaAllahYasser11 committed:
Merge branch 'EMS-12-forget-password-feature' into EMS-11-Login-feature

MennaAllahYasser11 committed:
Merge branch 'EMS-13-OAuth-setup' into EMS-11-Login-feature

mohamed17-a reviewed (approved):
Great work 👏 🚀
